packages/wasm-dpp/src/identity/identity_facade.rs

```diff
@@ -1,46 +1,278 @@
+use wasm_bindgen::__rt::Ref;
 
+#[derive(Clone)]
 #[wasm_bindgen(js_name=IdentityFacade)]
 pub struct IdentityFacadeWasm(IdentityFacade<BlsAdapter>);
 
-#[wasm_bindgen(js_class=IdentityFacade)]
 impl IdentityFacadeWasm {
-    #[wasm_bindgen(constructor)]
-    pub fn new(bls_adapter: JsBlsAdapter) -> IdentityFacadeWasm {
-        let bls = BlsAdapter(bls_adapter);
-        // TODO: REMOVE THAT LINE, TAKE IT AS AN ARGUMENT
-        let protocol_version_validator = ProtocolVersionValidator::default();
-        let public_keys_validator = PublicKeysValidator::new(bls).unwrap();
+    pub fn new(
+        protocol_version_validator: Arc<ProtocolVersionValidator>,
+        public_keys_validator: Arc<PublicKeysValidator<BlsAdapter>>,
+    ) -> IdentityFacadeWasm {
         let identity_facade = IdentityFacade::new(
-            Arc::new(protocol_version_validator),
-            Arc::new(public_keys_validator),
+            protocol_version_validator.protocol_version(),
+            protocol_version_validator,
+            public_keys_validator,
         )
         .unwrap();
 
         IdentityFacadeWasm(identity_facade)
     }
+}
 
-    #[wasm_bindgen()]
-    pub fn validate(
+#[wasm_bindgen(js_class=IdentityFacade)]
+impl IdentityFacadeWasm {
+    #[wasm_bindgen]
+    pub fn create(
         &self,
-        raw_identity_object: JsValue,
-    ) -> Result<ValidationResultWasm, NonConsensusErrorWasm> {
-        let identity_json = serde_wasm_bindgen::from_value(raw_identity_object)
-            .expect("unable to serialize identity");
-        // TODO: handle the case when
-        let validation_result = self.0.validate(&identity_json)?;
+        asset_lock_proof: JsValue,
+        public_keys: js_sys::Array,
+    ) -> Result<IdentityWasm, JsValue> {
+        let asset_lock_proof = create_asset_lock_proof_from_wasm_instance(&asset_lock_proof)?;
+
+        let raw_public_keys = to_vec_of_serde_values(public_keys.iter())?;
+
+        let public_keys = raw_public_keys
+            .into_iter()
+            .map(|v| IdentityPublicKey::from_raw_object(v).map(|key| (key.id, key)))
+            .collect::<Result<_, _>>()
+            .map_err(|e| format!("converting to collection of IdentityPublicKeys failed: {e:#}"))?;
+
+        self.0
+            .create(asset_lock_proof, public_keys)
+            .map(|identity| identity.into())
+            .map_err(from_dpp_err)
+    }
+
+    #[wasm_bindgen(js_name=createFromObject)]
+    pub fn create_from_object(
+        &self,
+        identity_object: JsValue,
+        options: JsValue,
+    ) -> Result<IdentityWasm, JsValue> {
+        let options: FromObjectOptions = if options.is_object() {
+            with_js_error!(serde_wasm_bindgen::from_value(options))?
+        } else {
+            Default::default()
+        };
+
+        let identity_json = utils::stringify(&identity_object)?;
+        let raw_identity: Value =
+            serde_json::from_str(&identity_json).map_err(|e| e.to_string())?;
```

Comment:
these two lines could be replaces with `raw_identity.with_serde_to_json_value()`  from `ToSerdeJsonExt`

Reply:
Fixed